src/consensus/blockchain.py

```diff
@@ -189,10 +194,10 @@ async def receive_block(
         """
         genesis: bool = block.sub_block_height == 0
 
-        if block.header_hash in self.sub_blocks:
+        if block.header_hash in self.__sub_blocks:
             return ReceiveBlockResult.ALREADY_HAVE_BLOCK, None, None
 
-        if block.prev_header_hash not in self.sub_blocks and not genesis:
+        if block.prev_header_hash not in self.__sub_blocks and not genesis:
```

Comment:
This does not support deep reorgs, this is very important. Let's say we cache the last 10k blocks, and the reorg is 30k blocks deep. Then we should load in blocks 10k at a time from the fork poinnt, etc. (Actually a whole epoch before the fork point, so we can calculate difficulty adjustments)

Reply:
Let me know if this doesn't make sennse

Reply:
no Im aware, but AFIK the sync uses the weight proofs to find forkpoints no

Reply:
@mariano54 ?